sql/logictest: TestParallel/subquery_retry_multinode failed

[(sql/logictest).TestParallel failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2713085&tab=buildLog) on [master@5cb9815fb8ef7635357ddaa4d4f606c9d2525a42](https://github.com/cockroachdb/cockroach/commits/5cb9815fb8ef7635357ddaa4d4f606c9d2525a42):

```
I210224 09:50:01.179936 11645862 testutils/testcluster/testcluster.go:135  [-] 2338 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:02.180131 11645862 testutils/testcluster/testcluster.go:135  [-] 2339  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:02.180131 11645862 testutils/testcluster/testcluster.go:135  [-] 2339 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:03.180354 11645862 testutils/testcluster/testcluster.go:135  [-] 2340  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:03.180354 11645862 testutils/testcluster/testcluster.go:135  [-] 2340 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:04.180640 11645862 testutils/testcluster/testcluster.go:135  [-] 2341  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:04.180640 11645862 testutils/testcluster/testcluster.go:135  [-] 2341 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:05.180891 11645862 testutils/testcluster/testcluster.go:135  [-] 2342  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:05.180891 11645862 testutils/testcluster/testcluster.go:135  [-] 2342 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:06.181136 11645862 testutils/testcluster/testcluster.go:135  [-] 2343  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:06.181136 11645862 testutils/testcluster/testcluster.go:135  [-] 2343 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:07.181359 11645862 testutils/testcluster/testcluster.go:135  [-] 2344  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:07.181359 11645862 testutils/testcluster/testcluster.go:135  [-] 2344 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:08.181596 11645862 testutils/testcluster/testcluster.go:135  [-] 2345  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:08.181596 11645862 testutils/testcluster/testcluster.go:135  [-] 2345 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:09.181860 11645862 testutils/testcluster/testcluster.go:135  [-] 2346  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:09.181860 11645862 testutils/testcluster/testcluster.go:135  [-] 2346 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:10.182117 11645862 testutils/testcluster/testcluster.go:135  [-] 2347  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:10.182117 11645862 testutils/testcluster/testcluster.go:135  [-] 2347 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:11.182334 11645862 testutils/testcluster/testcluster.go:135  [-] 2348  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:11.182334 11645862 testutils/testcluster/testcluster.go:135  [-] 2348 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210224 09:50:12.182581 11645862 testutils/testcluster/testcluster.go:135  [-] 2349  SucceedsSoon: unexpectedly found active spans:
I210224 09:50:12.182581 11645862 testutils/testcluster/testcluster.go:135  [-] 2349 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
    testcluster.go:135: condition failed to evaluate within 45s: unexpectedly found active spans:
             0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
        goroutine 11645862 [running]:
        runtime/debug.Stack(0xc0015fdbb0, 0x4eed140, 0xc040dc8b60)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x50b4500, 0xc004464c40, 0xc0015fdbb0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x6d
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).stopServers(0xc01da17200, 0x4f75f80, 0xc000074118)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:135 +0x23d
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).Start.func2()
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:335 +0x45
        github.com/cockroachdb/cockroach/pkg/util/stop.CloserFn.Close(0xc024816940)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:110 +0x25
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc007cd9a00, 0x4f75f80, 0xc000074110)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:478 +0x268
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*parallelTest).close(0xc004464c40)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/parallel_test.go:63 +0x88
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*parallelTest).run(0xc004464c40, 0xc00ed38060, 0x2f)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/parallel_test.go:175 +0x75b
        github.com/cockroachdb/cockroach/pkg/sql/logictest.TestParallel.func1(0xc000f28900)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/parallel_test.go:254 +0xb7
        testing.tRunner(0xc000f28900, 0xc02ae5de00)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
I210224 09:50:13.183089 11629234 sql/logictest/parallel_test.go:262  [-] 2350  2 parallel tests passed
    --- FAIL: TestParallel/subquery_retry_multinode (54.98s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=2

```
make stressrace TESTS=TestParallel PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestParallel.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestParallel failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2717492&tab=buildLog) on [master@26d86b709914e9a31ef92cfbf20d9c76ff8c149b](https://github.com/cockroachdb/cockroach/commits/26d86b709914e9a31ef92cfbf20d9c76ff8c149b):

```
I210225 09:06:12.138800 11842887 testutils/testcluster/testcluster.go:135  [-] 2314 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:13.139082 11842887 testutils/testcluster/testcluster.go:135  [-] 2315  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:13.139082 11842887 testutils/testcluster/testcluster.go:135  [-] 2315 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:14.139370 11842887 testutils/testcluster/testcluster.go:135  [-] 2316  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:14.139370 11842887 testutils/testcluster/testcluster.go:135  [-] 2316 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:15.139665 11842887 testutils/testcluster/testcluster.go:135  [-] 2317  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:15.139665 11842887 testutils/testcluster/testcluster.go:135  [-] 2317 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:16.139953 11842887 testutils/testcluster/testcluster.go:135  [-] 2318  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:16.139953 11842887 testutils/testcluster/testcluster.go:135  [-] 2318 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:17.140269 11842887 testutils/testcluster/testcluster.go:135  [-] 2319  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:17.140269 11842887 testutils/testcluster/testcluster.go:135  [-] 2319 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:18.140590 11842887 testutils/testcluster/testcluster.go:135  [-] 2320  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:18.140590 11842887 testutils/testcluster/testcluster.go:135  [-] 2320 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:19.140877 11842887 testutils/testcluster/testcluster.go:135  [-] 2321  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:19.140877 11842887 testutils/testcluster/testcluster.go:135  [-] 2321 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:20.141163 11842887 testutils/testcluster/testcluster.go:135  [-] 2322  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:20.141163 11842887 testutils/testcluster/testcluster.go:135  [-] 2322 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:21.141376 11842887 testutils/testcluster/testcluster.go:135  [-] 2323  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:21.141376 11842887 testutils/testcluster/testcluster.go:135  [-] 2323 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:22.141677 11842887 testutils/testcluster/testcluster.go:135  [-] 2324  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:22.141677 11842887 testutils/testcluster/testcluster.go:135  [-] 2324 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
I210225 09:06:23.141977 11842887 testutils/testcluster/testcluster.go:135  [-] 2325  SucceedsSoon: unexpectedly found active spans:
I210225 09:06:23.141977 11842887 testutils/testcluster/testcluster.go:135  [-] 2325 +     0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
    testcluster.go:135: condition failed to evaluate within 45s: unexpectedly found active spans:
             0.000ms      0.000ms    === operation:job _unfinished:1 intExec:create-stats
        goroutine 11842887 [running]:
        runtime/debug.Stack(0xc0021e3bb0, 0x4efa8a0, 0xc0928fd120)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x50c1f60, 0xc0069f7a40, 0xc0021e3bb0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x6d
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).stopServers(0xc090d5c900, 0x4f83aa0, 0xc000074118)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:135 +0x23d
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).Start.func2()
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:335 +0x45
        github.com/cockroachdb/cockroach/pkg/util/stop.CloserFn.Close(0xc04e818550)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:110 +0x25
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc03b76d380, 0x4f83aa0, 0xc000074110)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:478 +0x268
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*parallelTest).close(0xc0069f7a40)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/parallel_test.go:63 +0x88
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*parallelTest).run(0xc0069f7a40, 0xc01ce62060, 0x2f)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/parallel_test.go:175 +0x75b
        github.com/cockroachdb/cockroach/pkg/sql/logictest.TestParallel.func1(0xc009162600)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/parallel_test.go:254 +0xb7
        testing.tRunner(0xc009162600, 0xc0bfed6640)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
I210225 09:06:24.142351 11825909 sql/logictest/parallel_test.go:262  [-] 2326  2 parallel tests passed
    --- FAIL: TestParallel/subquery_retry_multinode (52.02s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=2

```
make stressrace TESTS=TestParallel PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestParallel.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/logictest/BUILD.bazel
+++ b/pkg/sql/logictest/BUILD.bazel
@@ -63,6 +63,7 @@ go_test(
         "//pkg/server",
         "//pkg/sql",
         "//pkg/sql/sessiondata",
+        "//pkg/sql/stats",
         "//pkg/sql/tests",
         "//pkg/testutils/serverutils",
         "//pkg/testutils/skip",

--- a/pkg/sql/logictest/parallel_test.go
+++ b/pkg/sql/logictest/parallel_test.go
@@ -34,6 +34,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/security"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/sql/stats"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
@@ -191,6 +192,8 @@ func (t *parallelTest) setup(spec *parTestSpec) {
 		st := server.ClusterSettings()
 		st.Manual.Store(true)
 		sql.DistSQLClusterExecMode.Override(&st.SV, int64(mode))
+		// Disable automatic stats - they can interfere with the test shutdown.
+		stats.AutomaticStatisticsClusterMode.Override(&st.SV, false)
 	}
 
 	t.clients = make([][]*gosql.DB, spec.ClusterSize)

